src/common/asb/Asb.c

```diff
@@ -629,6 +629,222 @@ static char* g_desiredEnsureDefaultDenyFirewallPolicyIsSet = NULL;
+    { "Ensure password reuse is limited", "57bdeae2-863e-14fb-a463-d6ee64816b33", "EnsurePasswordReuseIsLimited" },
+    { "Ensure the password hashing algorithm is SHA-512", "01ec5346-882b-485d-8960-01dedd608792", "EnsurePasswordHashingAlgorithm" },
+    { "Ensure the minimum days between password changes is configured", "50272a84-672d-4c11-a67e-9d058adaf67a", "EnsureMinDaysBetweenPasswordChanges" },
+    { "Ensure the maximum days between password changes is configured", "7F84F3A4-360D-494C-AA1E-1E15AE9FC9AC", "EnsureMaxDaysBetweenPasswordChanges" },
+    { "Ensure the inactive password lock period is configured", "91fbaeac-f5d0-4ac9-aa1b-52215aef1ed8", "EnsureInactivePasswordLockPeriod" },
+    { "Ensure all system accounts are non-login", "448b668a-738c-420b-b332-51ea49922933", "EnsureSystemAccountsAreNonLogin" },
+    { "Ensure default group for the root account is GID 0", "732fa92f-647e-47b8-b5a4-fdf00b02d9e2", "EnsureDefaultRootAccountGroupIsGidZero" },
+    { "Ensure root is the only UID 0 account", "b5845ff3-42f4-4112-b2a2-5b827232a053", "EnsureRootIsOnlyUidZeroAccount" },
+    { "Ensure unnecessary accounts are removed", "627b7494-0e62-4093-9f77-db8d526d036b", "EnsureUnnecessaryAccountsAreRemoved" },
+    { "Ensure the auditd package is installed", "f9fd03d2-75e4-4564-84a9-4e955f1e7c30", "EnsureAuditdInstalled" },
+    { "Ensure the auditd service is running", "c146c4a4-5eb6-4205-88da-5a71a82f2d45", "EnsureAuditdServiceIsRunning" },
+    { "Ensure the SNMP Server is not enabled", "ca1aea32-3969-49ab-abfc-2c5796a9a8bb", "EnsureSnmpServerIsDisabled" },
+    { "Ensure the rsync service is not enabled", "63aba603-b1f8-40df-82c5-38915452ce23", "EnsureRsynServiceIsDisabled" },
+    { "Ensure the NIS server is not enabled", "b4ad3fdd-7b68-4b11-a3ed-84b37a68b995", "EnsureNisServerIsDisabled" },
+    { "Ensure the rsh client is not installed", "6d441f31-f888-4f4f-b1da-7cfc26263e3f", "EnsureRshClientNotInstalled" },
+    { "Ensure SMB V1 with Samba is disabled", "7624efb0-3026-4c72-8920-48d5be78a50e", "EnsureSmbWithSambaIsDisabled" }
+};
+
+int AsbIsValidResourceIdRuleId(const char* resourceId, const char* ruleId, const char* payloadKey, void* log)
+{
+    int numRules = ARRAY_SIZE(g_rules);
+    int i = 0;
+    int result = 0;
+
+    if (NULL == payloadKey)
+    {
+        OsConfigLogError(log, "AsbIsValidRuleIdAndName: invalid payloadKey argument");
+        return EINVAL;
+    }
+
+    for (i = 0; i < numRules; i++)
+    {
+        if (0 == strcmp(payloadKey, (g_rules[i]).payloadKey))
+        {
+            if ((NULL != resourceId) && (0 != strcmp(resourceId, (g_rules[i]).resourceId)))
+            {
+                OsConfigLogError(log, "AsbIsValidRuleIdAndName: resourceId for rule '%s' of '%s' (instead of '%s') is invalid", payloadKey, resourceId, (g_rules[i]).resourceId);
+                result = ENOENT;
+            }
+            else if ((NULL != ruleId) && (0 != strcmp(ruleId, (g_rules[i]).ruleId)))
```

Comment:
Is it possible to use `strncmp` here as `RuleId` length is known?

Reply:
Good catch. Will do change.

Reply:
Fixed. Please see last iteration. Thanks for the feedback!